subprojects/ide/src/main/java/org/gradle/plugins/ide/api/GeneratorTask.java

```diff
@@ -52,6 +53,7 @@
  *
  * @param <T> The domain object for the configuration file.
  */
+@DisableCachingByDefault(because = "Base-class, not to be instantiated directly")
 public class GeneratorTask<T> extends ConventionTask {
```

Comment:
Good question. We don't use this class in our code, though there may be folks out there which use the task directly.

If we make the class abstract, will this break something? I suppose it won't, since
- You can use an abstract task class for registering a task
- Subclasses won't break since they this class does not have any abstract members.